Library/Homebrew/cmd/update-report.rb

```diff
@@ -69,6 +70,8 @@ def update_report
       updated = true
     end
 
+    migrate_legacy_cache_if_necessary
```

Comment:
It seems to me this method will keep running for each update if it fails. Noted that many fail reason won't be auto fixed if it's run second time.